docs/api_docs/matching.md

```diff
@@ -0,0 +1,55 @@
+# Matching API
+
+### Matching-related endpoints
+#### `GET intersection/liked-songs`
+- Returns json of the intersection of the current user's and the inputted user's liked songs playlists.
+- Status Codes:
+    - `200` - Successfully returned liked songs intersection
+- Sample `POST` body:
+```
+{
+    "target_user_id": 20
+}
+```
+- Sample response body:
+```
+{
+    "64rqvMhAPLLEag310IG3z9": {
+        "song": "Greek Tragedy - Oliver Nelson TikTok Remix",
+        "artist": "The Wombats / Oliver Nelson",
+        "album": "Greek Tragedy (Oliver Nelson TikTok Remix)",
+        "albumArtUrl": "https://i.scdn.co/image/ab67616d0000b2737f07fa3b19d425f7a9735111"
+    },
+    "0zQvlYjV4SM6tvvnpV3OFg": {
+        "song": "Weaponry",
+        "artist": "Mike Posner / Jessie J",
+        "album": "Operation: Wake Up",
+        "albumArtUrl": "https://i.scdn.co/image/ab67616d0000b273f066c94496426b9fe804a679"
+    }
+}
+```
+
+#### `GET user/update-liked-songs`
```

Comment:
Just brainstorming here... how are we going to handle stuff like this?

I don't think it's the best for the front-end to manually refresh everything, especially since it doesn't know anything about things like `audio-features-scores`, but I also don't know that it's efficient for the back-end to constantly ping and check for these things.

*(If we're not already...)* Could we maybe setup the server to automaticaly refresh these things on each login?

Reply:
I haven't really been looking at these endpoints so that may be more of a question for @RobBoeckermann but I do like the idea of refreshing on login and think that's probably the best way to do it.